fix(deps): upgrade cron-parser to v5.6.1

roggervalf opened the pull request:
<!--
  Thank you for submitting a PR! 
  Please fill out all sections below to help us understand your changes.
-->

### Why
<!-- 
  1. Why is this change necessary?
  2. What problem does it solve or improve?
  3. Link to any relevant issues, if applicable.
-->
 1. Why is this change necessary? Keep it up to date and fix https://github.com/taskforcesh/bullmq/issues/3252

### How
<!--
  1. How did you implement this?
  2. Outline the approach or steps taken.
  3. List any resources or documentation that helped you.
-->
_Enter the implementation details here._

### Additional Notes (Optional)
<!--
  Use this space for additional considerations: 
  - Potential side effects
  - Dependencies 
  - Testing instructions
  - Anything else reviewers should know
-->
fixes https://github.com/taskforcesh/bullmq/issues/3252

roggervalf committed:
fix(deps): upgrade cron-parser to v5.1.1

roggervalf committed:
docs: use link tag

roggervalf commented on src/interfaces/repeat-options.ts:
```diff
@@ -1,11 +1,20 @@
+  /**
+   * End date when job should stop
+   */
+  endDate?: Date | string | number;
```
I'm avoiding to reuse type from cron parser because it does not have descriptions and it also contains an extra custom type in their option attributes that is not compatible with Date class. Justo keeping the primitive types

manast commented on tests/test_job_scheduler.ts:
```diff
@@ -1645,7 +1645,7 @@ describe('Job Scheduler', function () {
         pattern: '0 1 * * *',
         endDate: new Date('2017-05-10 13:13:00'),
-        tz: 'Europe/Athens',
+        tz: 'Etc/UTC',
```
wouldn't this be a breaking change?

roggervalf replied:
I thought about it, no sure if aomeone is using utc option as we don't put it in our documentation. But if you prefer, it can be saved for next breaking change

manast committed:
refactor(backend)!: decouple Queue/Worker/Job from Redis via IQueueBackend

Introduce a datastore-agnostic `IQueueBackend` abstraction so the high-level
classes (Queue, Worker, Job, FlowProducer, QueueEvents, …) no longer depend on
`RedisClient`/`RedisConnection` directly. The backend now owns its connection(s)
and is injected through a `BackendFactory` (default `createRedisBackend`),
clearing the way for non-Redis backends.

- Add `IQueueBackend` (src/interfaces/queue-backend.ts): every high-level data
  op plus connection lifecycle (waitUntilReady/close/disconnect/setName,
  on/once/removeListener for ready|error|close, minimumBlockTimeout,
  waitForJob, readEvents, disconnect/reconnectBlocking, forQueue, closing).
- Add `BackendFactory` type and inject it into the constructors of QueueBase,
  Queue, Worker, JobScheduler, QueueEvents, QueueEventsProducer and
  FlowProducer (defaulting to `createRedisBackend`).
- Rename `Scripts` -> `RedisQueueBackend` (the Redis adapter); it now extends
  EventEmitter, owns its `RedisConnection`(s), encapsulates the blocking
  connection, and owns the blocking `waitForJob` (capability-based timeout
  rounding + stuck-connection watchdog moved out of Worker).
- Add `createRedisBackend` factory (src/utils/create-backend.ts) that builds and
  owns the main and (optional) dedicated blocking connection.
- FlowProducer spans multiple queues over a single connection via
  `IQueueBackend.forQueue(queueName)`, returning a sibling backend bound to the
  node's keys that shares the connection.
- Remove the legacy repeatable-jobs implementation in favour of job schedulers:
  delete the `Repeat` class and the `repeat` option on `add`/`addBulk`; the
  internal repeat carrier now lives in `JobSchedulerJobOptions`.
- Slim `MinimalQueue`/`ScriptQueueContext`; the latter is now internal to the
  Redis adapter only.

Tests: replace removed escape hatches with test helpers
(`tests/utils/get-redis-client.ts`: getRedisClient/getBlockingRedisClient/
getRedisVersion/getDatabaseType/getRedisConnection/getBlockingRedisConnection).
Stabilize two pre-existing fake-timer scheduler tests (stub `worker.delay`,
make the delayed job due before starting the worker / use `autorun: false`).

BREAKING CHANGE: the high-level classes no longer expose Redis internals. The
optional `Connection` constructor parameter is replaced by an optional
`BackendFactory`. `Queue#client`, `Queue#redisVersion`, `Queue#databaseType`,
`Worker#blockingClient` and `FlowProducer#client` are removed; reach the raw
Redis client via the concrete `RedisQueueBackend` returned by `getBackend()`.
`Worker#waitUntilReady()` now resolves to `void` instead of the client.

Legacy repeatable jobs and their API are no longer supported. The `repeat`
option on `Queue#add`/`addBulk` and the `Repeat` class are removed, along with
`Queue#getRepeatableJobs`, `Queue#removeRepeatable` and
`Queue#removeRepeatableByKey`. Use job schedulers (`Queue#upsertJobScheduler`,
`Queue#getJobSchedulers`, `Queue#removeJobScheduler`) instead.

manast committed:
fix: address PR comments

manast committed:
fix: address PR comments

manast committed:
fix: address PR comments

manast committed:
docs: add high level documentation for the backend interface

roggervalf committed:
chore: fix merge conflicts

roggervalf committed:
Merge branch 'refactor/refactor-redis-backend' into cron-parser

roggervalf committed:
chore: update lock

roggervalf committed:
Merge branch 'master' into cron-parser

roggervalf committed:
chore: update lock file

roggervalf committed:
Merge branch 'feat/v6' into cron-parser

roggervalf committed:
ci: consider feat/v6 branch when executing test cases

roggervalf committed:
chore: fix merge conflicts

roggervalf committed:
test: fix test case

Copilot committed:
Merge remote-tracking branch 'origin/feat/v6' into cron-parser

# Conflicts:
#	src/interfaces/queue-options.ts
#	yarn.lock

Copilot committed:
fix: add return type annotation to resolve TS7011 error in queue.ts
